### PR TITLE
Modify makefiles and `gcc-backend` to propagate LDLIBS.

### PR DESCRIPTION
**Pull request by JoshDreamland**
This will be important for the test harness.

### CODE REVIEW

File: CompilerSource/compiler/compile.cpp [hunk 1]
```diff
@@ -607,18 +607,11 @@ wto << "namespace enigma_user {\nstring shader_get_name(int i) {\n switch (i) {\
...
-  }
-  else {
-    if (TOPLEVEL_cflags.length()) make += "CFLAGS=\"" + TOPLEVEL_cflags + " -g -DDEBUG_MODE\" ";
-    if (TOPLEVEL_cppflags.length()) make += "CPPFLAGS=\"" + TOPLEVEL_cppflags + "\" ";
-    if (TOPLEVEL_cxxflags.length()) make += "CXXFLAGS=\"" + TOPLEVEL_cxxflags + " -g -DDEBUG_MODE\" ";
-    if (TOPLEVEL_ldflags.length()) make += "LDFLAGS=\"" + TOPLEVEL_ldflags + "\" ";
-  }
+  if (TOPLEVEL_cflags.length()) make += "CFLAGS=\"" + TOPLEVEL_cflags + "\" ";
+  if (TOPLEVEL_cppflags.length()) make += "CPPFLAGS=\"" + TOPLEVEL_cppflags + "\" ";
+  if (TOPLEVEL_cxxflags.length()) make += "CXXFLAGS=\"" + TOPLEVEL_cxxflags + "\" ";
```
Comment (JoshDreamland) [inline CompilerSource/compiler/compile.cpp]: Where's the `-g` coming from, now, @fundies?

Comment (fundies) [reply CompilerSource/compiler/compile.cpp]: its in the shellmain makefile

Comment (JoshDreamland) [reply CompilerSource/compiler/compile.cpp]: Good enough for me. Thanks for making this change.

**Comment by RobertBColton** [comment]
Whatever this is, I am not sure if it slowed down building the compiler. I just merged upstream in locally and I deleted and rebuilt compileEGMf.dll using the old git-bash.bat from the portable ZIP by calling make inside `enigma-dev` and it took like forever. Then I deleted it again and launched LGM then ran a game and it was real fast to build. So I don't know if it slowed it down or not.

**Pull request merged**